internal/kubernetes/operator/deployment/deployment.go

```diff
@@ -196,7 +196,7 @@ func buildGlobalDeployment(atlasRepSpec []*mongodbatlas.AdvancedReplicationSpec,
 			Db:                     ns.Db,
 			Collection:             ns.Collection,
 			CustomShardKey:         ns.CustomShardKey,
-			NumInitialChunks:       ns.NumInitialChunks,
+			NumInitialChunks:       int(*ns.NumInitialChunks),
```

Comment:
It is recommended use getters from sdk and avoid using * to not get any memory allocation errors.

Reply:
Oh, will change this. Thanks Wojciech!